substrate/zombienet/0001-basic-warp-sync/README.md

```diff
@@ -10,17 +10,18 @@ The `dave` node executed with `--sync warp` syncs database with the rest of the
 Database was prepared using the following zombienet file (`generate-warp-sync-database.toml`):
 ```
 [relaychain]
-default_image = "docker.io/parity/substrate:master"
+default_image = "docker.io/paritypr/substrate:master"
 default_command = "substrate"
 
-chain = "gen-db"
+chain = "local"
 
 chain_spec_path = "chain-spec.json"
 
   [[relaychain.nodes]]
   name = "alice"
   validator = true
-
+  args = ["--state-pruning archive --blocks-pruning archive"]
```

Comment:
why is this needed? to my understanding we should need neither state nor block bodies for warp syncing..

Reply:
I'm not sure why it's needed, but without it the nodes from which the warp proof is downloaded generate some errors that go something like `State for block ... already discarded` when they are requested for warp sync fragments. And the warp sync freezes. But good point. I can investigate this separately. Will take a note.

Reply:
Do you see this log `State for block ... already discarded` every time?
I was investigating that some time ago, and it was happening sometimes (10%, maybe 20% depending on CPU load):
https://github.com/paritytech/polkadot-sdk/issues/2568

When preparing then database snapshot for warp-sync test we should keep all blocks, so adding this option could be desirable.

Reply:
> Do you see this log `State for block ... already discarded` every time?

Yes, for me this was happening very often. Definitely more than 50% of the runs. Maybe also because the new db is bigger.

Reply:
Sorry, second thought: we don't actually need this params. It will indirectly fix #2568, but the problem will still be there. The database actually does not need to be full archive.

Reply:
Better workaround would be to forbid Bob to connect to Alice.

Reply:
Sounds good. Do you know how this can be done ? IS there a config for it ? And would this prevent bob to connect to alice just for getting the warp sync ? Would they still be able to connect to each other in order to produce blocks ?

Reply:
Thanks @skunert ! Will try.

Reply:
> Please also re-upload db-snapshot created w/o archive mode.

@michalkucharczyk I just checked this and indeed it solves the warp sync issue, but the chain isn't finalizing blocks because bob can't talk to alice. Is it ok if I leave the archide db for the moment ? I added a comment that this is a workaround and that it should be removed after fixing https://github.com/paritytech/polkadot-sdk/issues/2568

Reply:
@michalkucharczyk sorry for the ping. WDYT ? Would you have any concern about merging this PR as it is and then switching to a pruned snapshot in a future PR? After https://github.com/paritytech/polkadot-sdk/issues/2568 is fixed ?

Reply:
I'd prefer to have DB without archive mode (because this is more close to scenario used widely in production).

But yeah, let's use archive DB here. I have some ideas for fixing #2568. Please also leave comment there (just not to forget about this). I would also appreciate if you leave the actual toml file that was used for generating DB in repo  (with appropriate comment).

Thank you.

Reply:
Ok thanks ! Done:

- Added comment to #2568 here: https://github.com/paritytech/polkadot-sdk/issues/2568#issuecomment-1886745731
- The toml file that was used for generating DB is already updated and it also has a TODO comment about removing the workaround

Reply:
ty!